server/ValidateGlobalRules.java

```diff
@@ -427,4 +436,44 @@ private static ReasonerQuery combinedRuleQuery(ReasonerQueryFactory reasonerQuer
         }
         return Optional.empty();
     }
+
+    /**
+     * We have an over-strict defensive check we enforce for now, to ensure that a rule will not throw at runtime
+     * Requiring a '!=' between variables that may result in an inferred relation with duplicate role player edges.
+     * For example:
+     *
+     *
+     * @param headQuery
+     * @param bodyQuery
+     * @return
+     */
+    private static Set<String> validateDuplicateInferredRolesMustBeDifferentConcepts(ReasonerQuery headQuery, ReasonerQuery bodyQuery) {
```

Comment:
it might be better to put this check in `RelationAtomValidator` and call it in `validateAsRuleHead`